Sources/NIO/Interfaces.swift

```diff
@@ -206,6 +227,57 @@ extension NIONetworkDevice {
         /// This constructor will fail if NIO does not understand the format of the underlying
         /// socket address family. This is quite common: for example, Linux will return AF_PACKET
         /// addressed interfaces on most platforms, which NIO does not currently understand.
+#if os(Windows)
+        internal init?(_ pAdapter: UnsafeMutablePointer<IP_ADAPTER_ADDRESSES>,
+                       _ pAddress: UnsafeMutablePointer<IP_ADAPTER_UNICAST_ADDRESS>) {
+            self.name = String(decodingCString: pAdapter.pointee.FriendlyName,
+                               as: UTF16.self)
+            self.address = pAddress.pointee.Address.lpSockaddr.convert()
+
+            let v4mask: (UINT8) -> SocketAddress? = { _ in
+                var buffer: [CChar] =
+                    Array<CChar>(repeating: 0, count: Int(INET_ADDRSTRLEN))
+                var mask: sockaddr_in = sockaddr_in)
+                mask.sin_family = ADDRESS_FAMILY(BSDSocket.AF_INET)
+                _ = buffer.withUnsafeMutableBufferPointer {
+                    try! BSDSocket.inet_ntop(af: BSDSocket.AF_INET, src: &mask,
+                                             dst: $0.baseAddress!,
+                                             size: INET_ADDRSTRLEN)
+                }
+                return SocketAddress(mask, host: mask.addressDescription())
+            }
+            let v6mask: (UINT8) -> SocketAddress? = { _ in
+                var buffer: [CChar] =
+                    Array<CChar>(repeating: 0, count: Int(INET6_ADDRSTRLEN))
+                var mask: sockaddr_in6 = sockaddr_in6()
+                mask.sin6_family = ADDRESS_FAMILY(BSDSOcket.AF_INET6)
+                _ = buffer.withUnsafeMutableBufferPointer {
+                    try! BSDSocket.inet_ntop(af: BSDSocket.AF_INET6, src: &mask,
+                                             dst: $0.baseAddress!,
+                                             size: INET6_ADDRSTRLEN)
+                }
+                return SocketAddress(mask, host: mask.addressDescription())
+            }
+
+            switch pAddress.pointee.Address.lpSockaddr.pointee.sa_family {
+            case ADDRESS_FAMILY(AF_INET):
+                self.netmask = v4mask(pAddress.pointee.OnLinkPrefixLength)
+                self.interfaceIndex = Int(pAdapter.pointee.IfIndex)
+                break
+            case ADDRESS_FAMILY(AF_INET6):
+                self.netmask = v6mask(pAddress.pointe.OnLinkPrefixLength)
+                self.interfaceIndex = Int(pAdapter.pointee.Ipv6IfIndex)
+                break
+            default:
+                nil
+            }
+
+            // TODO(compnerd) handle broadcast/ppp/multicast information
+            self.broadcastAddres = nil
```

Comment:
This won't compile.

Reply:
Im not getting any warnings for this atm.  Am I missing something?

Reply:
There's an `s` missing.

Reply:
Impressive, I managed to miss that upon multiple readings.